ai_feedback/code_processing.py

```diff
@@ -56,20 +57,17 @@ def process_code(args, prompt: str) -> Tuple[str, str]:
             )
         ]
 
-    for file in assignment_files:
-        filename = os.path.basename(file)
-        name_without_ext, _ = os.path.splitext(filename)
-
-        if name_without_ext.endswith("_solution"):
-            prompt += (
-                f"\nThe instructor's solution file you should reference is {filename}."
-            )
-        elif name_without_ext.endswith("_submission"):
-            prompt += f"\nThe student's code submission file you should reference is {filename}."
-        elif name_without_ext.endswith("test_output"):
-            prompt += (
-                f"\nThe student's error trace file you should reference is {filename}."
-            )
+    # Use template rendering system for all prompt building
```

Comment:
Overall, I think you can move all of the logic for search for template components into `render_prompt_template`, so that in this function you just call `render_prompt_template` and pass in the necessary arguments. This will help reduce the code duplication between here and the other place where `render_prompt_template` is called.